### PR TITLE
add more data type to samples linking - SAM-4

### PR DESCRIPTION
**Pull request by Tianhao-Gu**

### CODE REVIEW

File: test/data/fake_samples.tsv [hunk 1]
```diff
@@ -2,4 +2,4 @@ validator types:	number	noop	onto	enum	onto	user field	number
 test name field	Depth bgs	Country	biome	material	feature	user field	elevation start	elevation unit	size
 s1	0	Germany	ENVO:01000221	Gas	ENVO:01000002		123	cm	1
 s2	1.04	Uzbekistan		Soil		blip	3	feet	1
-s3	299	Cambodia	temperate woodland biome		ENVO:00002226	blap blork	5.9	meters	1
```
Comment (slebras) [inline test/data/fake_samples.tsv:5]: I think we should leave this as "temperate woodland biome", its to test the functionality of looking up the term id given the term name from the OntologyAPI, which I think we added support for on certain terms

Comment (Tianhao-Gu) [reply test/data/fake_samples.tsv]: 👍

File: ui/narrative/methods/link_samples/spec.json [hunk 2]
```diff
@@ -94,13 +98,6 @@
                     "input_parameter": "sample_set_ref",
                     "target_property": "sample_set_ref",
                     "target_type_transform": "resolved-ref"
-                },{
-                    "input_parameter": "sample_name",
```
[the remark this answers is not in the document]
Comment (Tianhao-Gu) [reply ui/narrative/methods/link_samples/spec.json:98]: right. The `sample_name` and `obj_ref` (used to be `reads_ref`) are not arguments to the `link_samples` function. They are 'stored' in the `links` argument. All the func needs is the `links` argument. I think they shouldn't be there in the first place.

**Comment by Tianhao-Gu** [comment]
@slebras I reversed the `fake_samples.tsv` test file. Our tests will fail due to errors in OntologyAPI.

File: ui/narrative/methods/link_samples/spec.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (slebras) [reply ui/narrative/methods/link_samples/spec.json:98]: ah okay, gotcha

**Pull request merged**